walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,98 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
-}
+  private static final Collection<Event> NO_EVENTS = Collections.emptySet();
+  private static final Collection<String> NO_ATTENDEES = Collections.emptySet();  
+  private static final int endOfDay = TimeRange.END_OF_DAY; 
+  
+  private static int WHOLE_DAY = 1440;
+
+    // sort and return busy times for requested attendees
+    private static Collection<TimeRange> getCoveredTimesForAttendees(Collection<Event> events,
+    Collection<String> mandatoryAttendees) {   
+      Set<String> setOfAttendees = new HashSet<String>();
+      for(String attendee : mandatoryAttendees) {
+        setOfAttendees.add(attendee);
+      }
+
+      // define 'coveredTimesForAttendees' as ArrayList to use 
+      // suitable method for sorting the time ranges
+      ArrayList<TimeRange> coveredTimesForAttendees = new ArrayList <TimeRange> ();
```

Comment:
nit: ArrayList<TimeRange>();

No need for spaces.

Reply:
Done

Reply:
Done!